docs/security/access-restriction-with-fail2ban.md

```diff
@@ -0,0 +1,325 @@
+Access restriction with Fail2ban on CentOS/Debian
+
+Any Internet-connected server will be at some point the target of unauthorized access attempts. While these attacks are usually harmless, they are a nuisance: system resources are being used to serve the unauthorized access attempts and system logs increase in size and become cluttered. Even worse, any public-facing accounts (FTP, POP3/IMAP, SSH) with weak passwords can be accessed by a thorough password scanner. 
+
+This document will explain how to further restrict access to your Debian/CentOS server using Fail2ban, a tool which can trigger actions (most of the time, the action is to ban the offender's IP address) based on events logged in your system (a number of failed attempts to log into a service, for example).
+
+# Prerequisites
+
+Ensure your system is up to date before starting. 
+
+For CentOS, run:
+
+	[root@server ~]# yum update
+
+For Debian, run:
+
+	[root@server ~]# apt-get update; apt-get upgrade
+
+## CentOS 
+
+Fail2ban is provided by the EPEL repository, so this needs to be installed first:
+
+	[root@server ~]# yum -y install epel-release
+
+Update the yum repository and install fail2ban:
+
+	[root@server ~]# yum update; yum -y install fail2ban
+
+## Debian
+
+fail2ban is already included in the standard distribution, so you just need to run:
+
+	[root@server ~]# apt-get install fail2ban
+
+The configuration set for a specific application is called a "jail" and contains the jail settings, the filter and the required action. Fail2ban will permanently monitor the specified log file using the filter and jail settings and trigger the action if the conditions are matched. After a configurable time, another action can be performed (such as unbanning the offending IP).
+
+# Fail2ban configuration
+
+## Configuration files
+
+Configuration files are located under `/etc/fail2ban`. You will find here:
+
+- `jail.conf`: the stock jails configuration, which for most users will be sufficient. Should you wish to create your own jails or adjust the existing jails' configuration, it's recommended to deploy them in a "jail.local" file and leave the original file untouched. Otherwise, your jail.conf customizations might get lost when upgrading fail2ban.
+- `fail2ban.conf`: the general fail2ban configuration file. Same as above, any customizations should be placed in a separate file named fail2ban.local.
+- `filter.d`: the folder containing built-in (predefined) filters, each filter being in a separate file.
+- `action.d`: the folder containing built-in (predefined) actions, each action being in a separate file.
+
+## Starting and stopping fail2ban
+
+The fail2ban service is started/stopped/checked using the standard commands:
+
+	[root@server ~]# /etc/init.d/fail2ban status
```

Comment:
As these are separate commands, they should be in separate code blocks, under distinct steps. Also, on Debian 8 you can't see the currently loaded jails from the status output:

```
# /etc/init.d/fail2ban status
● fail2ban.service - LSB: Start/stop fail2ban
   Loaded: loaded (/etc/init.d/fail2ban)
   Active: active (running) since Tue 2015-06-02 16:13:07 EDT; 2min 8s ago
   CGroup: /system.slice/fail2ban.service
           └─3476 /usr/bin/python /usr/bin/fail2ban-server -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid
```